robottelo/test.py

```diff
@@ -101,9 +102,148 @@
+            else False
+        )
+
+    @property
+    def value_name(self):
+        """Returns CLI expected value name (return code)"""
+        return 'return code'
+
+
+class _AssertNotRaisesContext(object):
+    """A context manager used to implement :meth:`TestCase.assertNotRaises`.
+    """
+
+    def __init__(self, expected, test_case, expected_regex=None,
+                 expected_value=None, value_handler=None):
+        self.expected = expected
+        self.expected_regex = expected_regex
+        self.value_handler = value_handler
+        if value_handler is None and expected_value:
+            self.value_handler = test_case._default_notraises_value_handler(
+                expected_value)
+        self.failureException = test_case.failureException
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, exc_value, tb):
+        if exc_type is None:
+            return True
+
+        try:
+            exc_name = self.expected.__name__
+        except AttributeError:
+            exc_name = str(self.expected)
+
+        self.exception = exc_value  # store for later retrieval
+
+        if issubclass(exc_type, self.expected):
+            if not any((self.value_handler, self.expected_regex)):
+                raise self.failureException(
+                    "{0} raised".format(exc_name))
+            regex = self.expected_regex
+            response_code = None
+            if self.value_handler:
+                response_code = self.value_handler.validate(exc_value)
+            if regex:
+                regex = True if regex.search(str(exc_value)) else False
+
+            if response_code and regex:
+                raise self.failureException(
+                    "{0} raised with {1} {2} and {3} found in {4}"
+                    .format(
+                        exc_name,
+                        self.value_handler.value_name,
+                        self.value_handler.expected_value,
+                        self.expected_regex.pattern,
+                        str(exc_value),
+                    )
+                )
+            elif response_code and regex is None:
+                raise self.failureException(
+                    "{0} raised with {1} {2}".format(
+                        exc_name,
+                        self.value_handler.value_name,
+                        self.value_handler.expected_value,
+                    )
+                )
+            elif regex and response_code is None:
+                raise self.failureException(
+                    "{0} raised and {1} found in {2}".format(
+                        exc_name,
+                        self.expected_regex.pattern,
+                        str(exc_value),
+                    )
+                )
+            else:
+                # pass through
+                return False
+
+        else:
+            # pass through
+            return False
+
+
```

Comment:
`_AssertNotRaisesContext` is more related to `assertNotRaises` rather to value helpers, and without `_AssertNotRaisesContext` - it's just two small classes (which are closely connected to APITestCase and CLITestCase here btw). I'm not sure if they worth separate module (it's just ~50 lines) but i can move them if you think it's worth it.